lib/spack/spack/cmd/audit.py

```diff
@@ -61,9 +69,10 @@ def configs(parser, args):
 
 
 def packages(parser, args):
-    pkgs = args.name or spack.repo.PATH.all_package_names()
-    reports = spack.audit.run_group(args.subcommand, pkgs=pkgs)
-    _process_reports(reports)
+    with spack.audit.strict_variants():
```

Comment:
This needs a check for whether `args.strict_variants` is True to determine whether to use this or `nullcontext`.

Reply:
fixed